From: Peter Stokes <linux@dadeos.co.uk>
Date: Mon, 12 Jan 2009 11:29:41 +0000 (+0100)
Subject: Fix wrong direction check in aplay/arecord --list-pcms
X-Git-Url: http://git.alsa-project.org/?p=alsa-utils.git;a=commitdiff_plain;h=9a47cf7b35f9840379f6ebdcf429128ddc572049

Fix wrong direction check in aplay/arecord --list-pcms

I have just discovered a minor logic inversion bug in
aplay/arecord --list-pcms functionality.

Basically, executing "aplay --list-pcms" lists all devices capable of
capture and executing "arecord --list-pcms" lists all devices capable
of playback.

Signed-off-by: Peter Stokes <linux@dadeos.co.uk>
Signed-off-by: Takashi Iwai <tiwai@suse.de>
---

diff --git a/aplay/aplay.c b/aplay/aplay.c
index 2e14d2a..837e46a 100644
--- a/aplay/aplay.c
+++ b/aplay/aplay.c
@@ -287,7 +287,7 @@ static void pcm_list(void)
 		name = snd_device_name_get_hint(*n, "NAME");
 		descr = snd_device_name_get_hint(*n, "DESC");
 		io = snd_device_name_get_hint(*n, "IOID");
-		if (io != NULL && strcmp(io, filter) == 0)
+		if (io != NULL && strcmp(io, filter) != 0)
 			goto __end;
 		printf("%s\n", name);
 		if ((descr1 = descr) != NULL) {
